src/exit_codes.rs

```diff
@@ -7,9 +7,21 @@ pub enum ExitCode {
 impl Into<i32> for ExitCode {
     fn into(self) -> i32 {
         match self {
-            ExitCode::Success => 0,
-            ExitCode::GeneralError => 1,
-            ExitCode::KilledBySigint => 130,
+            Self::Success => 0,
+            Self::GeneralError => 1,
+            Self::KilledBySigint => 130,
         }
     }
 }
+
+impl ExitCode {
+    pub fn error_if_any_error(results: Vec<Self>) -> Self {
```

Comment:
It feels a little strange to me to have this in an `impl ExitCode` block. I think it could easily be a free function.

As for the naming, maybe: `merge_exitcodes`?

Finally: can we please add a few unit tests for this function (in this file)?

Reply:
I think I've added a few interesting unit tests of `merge_exitcodes`.

Let us know if they're good enough 😄